proposed/project-file-simplification.md

```diff
@@ -0,0 +1,351 @@
+
+```bash
+dotnet project upgrade 11.0.100
+```
+
+This would update the project file with the requested `Base` and new properties based on a _diff_ between the previous and resulting project properties.
+
+For example, if the 11.0.100 version of `Base` had two new enabled properties `UseFeatureA` and `UseFeatureB`, two new properties would be added to the project file when upgrading from a version 10 to version 11:
+
+```XML
+<Project Base="10.0.100" 
+         Type="Console"> 
+</Project> 
+```
+
+```XML
+<Project Base="11.0.100" 
+         Type="Console">
+    <PropertyGroup Label="These properties added on upgrade from 10.0.100 to 11.0.100"
+       <UseFeatureA>disable</UseFeatureA>
+       <UseFeatureB>disable</UseFeatureB>
+    </PropertyGroup>
+</Project> 
+```
+
+This approach is simple for a project file. It might be extended to complex repos by placing the `PropertyGroup` containing the new properties at the start of whichever file (project file, `Directory.Build.props` or `Import` file) the `Base` attribute appeared in. However, the command would probably be run in the same directory as the file containing the `Base` attribute and specify the file if there were multiple props files.
+
+#### Interactive upgrade
+
+|Command|Result|
+|-|-|
+|dotnet project upgrade 11.0.100 --whatif|List the properties that would be added, along with any inline comments|
+|dotnet project upgrade 11.0.100 --preserve|Actually run the upgrade, resulting in new properties to match previous behavior|
+|dotnet project upgrade 11.0.100 --no-preserve|Actually run the upgrade, without new properties|
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+This introduces a new concept to .NET users, and some additional complexity in the overall MSBuild evaluation. Users would see this in areas such as understanding binlogs. We think the trade off is worthwhile because users get small clean project files the ability to control new warnings/breaking changes independent of both SDK version and TargetFramework.
+
+## Alternatives
+[alternatives]: #alternatives
+
+We could continue to tie new features to TFM. THe drawback is that users generally upgrade their TFM to keep their projects in support. While this avoids breaking changes/warnings occurring on tooling upgrade, it may either delay adopting desirable new features, or result in breaking changes/warnings happening when the user is just trying to stay in support.
+
+## Open questions
+[open]: #open-questions
+
+### What capacity to pin and float should we offer for `Base` versions?
+
+### What about multi-targeting?
+
+### Should we restrict `Type` to the project file?
+
+Stated differently, should we allow `Type` to be in  `Directory.Build.props`. Since `Type` determines the SDK and that contains the targets that run most build operations, this is tricky. Also, it may be best for users to see this in the project file for clarity.
```

Comment:
```suggestion
Stated differently, should we allow `Type` to be in `Directory.Build.props`? Since `Type` determines the SDK and that contains the targets that run most build operations, this is tricky. Also, it may be best for users to see this in the project file for clarity.
```